proto_parser/src/proto_value.cc

```diff
@@ -26,8 +26,20 @@ std::string ProtoValue::PrintToTextProto() {
 
 std::unique_ptr<ProtoValue> ProtoValue::Create(absl::string_view text_proto, 
     google::protobuf::Message& message) {
-  // TODO(alexanderlin): Implement.
-  return std::unique_ptr<ProtoValue>();
+  google::protobuf::TextFormat::Parser parser;
+  google::protobuf::TextFormat::ParseInfoTree tree;
+  parser.WriteLocationsTo(&tree);
+  parser.ParseFromString(std::string(text_proto), &message);
+  ProtoParser proto_parser(text_proto);
+  
+  int last_field_loc = 0;
```

Comment:
Passing primitive type by reference (e.g. int&) is not a good practice.

If you really want to pass last_field_loc, please use a struct to wrap it, like

struct XX {
  int last_field_loc = 0;
}

and you can pass XX* into functions.